scripts/e2e-cluster-dump.sh

```diff
@@ -79,96 +79,116 @@ function cluster-describe {
     kubectl -n mayastor describe daemonsets
 }
 
-function logs-csi-containers {
-    mayastor_csipods=$(kubectl -n mayastor get pods | grep mayastor-csi | sed -e 's/ .*//')
-    for pod in $mayastor_csipods
-    do
-        echo "# $pod csi-driver-registrar $* ---------------------------------"
-        kubectl -n mayastor logs "$@" "$pod" csi-driver-registrar
-    done
+function hasPodRestarted {
+    rst=$(kubectl -n mayastor get pods "$1" | grep -v NAME | awk '{print $4}')
+    return $((rst))
+}
 
-    moacpod=$(kubectl -n mayastor get pods | grep moac | sed -e 's/ .*//')
-    echo "# $moacpod csi-provisioner $* ---------------------------------"
-    kubectl -n mayastor logs "$@" "$moacpod" csi-provisioner
-    echo "# $moacpod csi-attacher $* ---------------------------------"
-    kubectl -n mayastor logs "$@" "$moacpod" csi-attacher
+# args filename kubectlargs
+# filename == "" -> stdout
+function kubectlEmitLogs {
+    fname=$1
+    shift
+
+    if [ -n "$fname" ]; then
+        kubectl -n mayastor logs "$@" >& "$fname"
+    else
+        kubectl -n mayastor logs "$@"
+    fi
 }
 
-function logs-csi-mayastor {
+# args = destdir podname containername
+# if $destdir != "" then a log files are generate in $destdir
+#   with the name of the pod and container.
+function emitPodContainerLogs {
+    destdir=$1
+    podname=$2
+    containername=$3
+
+    if [ -z "$podname" ] || [ -z "$containername" ]; then
+        echo "ERROR calling emitPodContainerLogs"
+        return
+    fi
+
+    if ! hasPodRestarted "$podname" ; then
+        if [ -z "$destdir" ]; then
+            echo "# $podname $containername previous -------------------"
+            logfile=""
+        else
+            logfile="$destdir/$podname.$containername.previous.log"
+        fi
+
+        kubectlEmitLogs "$logfile" -p "$podname" "$containername"
+    fi
+
+    if [ -z "$destdir" ]; then
+        echo "# $podname $containername ----------------------------"
+        logfile=""
+    else
+        logfile="$destdir/$podname.$containername.log"
+    fi
+
+    kubectlEmitLogs "$logfile" "$podname" "$containername"
+}
+
+# arg1 = destdir or "" for stdout
+function getLogsMayastorCSI {
     mayastor_csipods=$(kubectl -n mayastor get pods | grep mayastor-csi | sed -e 's/ .*//')
     for pod in $mayastor_csipods
     do
-        echo "# $pod mayastor-csi $* ---------------------------------"
-        kubectl -n mayastor logs "$@" "$pod"  mayastor-csi
+        # args destdir podname containername
```

Comment:
For consistency, should we move this comment above the function like on line 100?
Same for lines 150 & 158.

Reply:
Ah, no the intention was to aid readability by describing  the arguments for the `emitPodContainerLogs` function, I will replace `args` with  `emitPodContainerLogs` in the comments.